sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredential.java

```diff
@@ -9,18 +9,56 @@
+ * <li>{@link ManagedIdentityCredential} - If the application deploys to an Azure host with Managed Identity enabled, the DefaultAzureCredential will authenticate with that account.</li>
+ * <li>{@link IntelliJCredential} - If you've authenticated via Azure Toolkit for IntelliJ, the DefaultAzureCredential will authenticate with that account.</li>
+ * <li>{@link AzureCliCredential} - If you've authenticated an account via the Azure CLI az login command, the DefaultAzureCredential will authenticate with that account.</li>
+ * <li>{@link AzurePowerShellCredential} - If you've authenticated an account via the Azure Power Shell Az Login command, the DefaultAzureCredential will authenticate with that account.</li>
  * <li>Fails if none of the credentials above could be created.</li>
  * </ol>
+ *
+ * For more information refer to <a href="https://aka.ms/azsdk/java/identity/defaultazurecredential/docs"> Conceptual knowledge and configuration details </a>.
+ *
+ * <H2>Configure DefaultAzureCredential</H2>
+ * DefaultAzureCredential supports a set of configurations through setters on the DefaultAzureCredentialBuilder or environment variables.
+ * <ol>
+ *     <li>Setting the environment variables AZURE_CLIENT_ID, AZURE_CLIENT_SECRET/AZURE_CLIENT_CERTIFICATE_PATH, and AZURE_TENANT_ID configures the DefaultAzureCredential to authenticate as the service principal specified by the values.</li>
+ *     <li>Setting {@link DefaultAzureCredentialBuilder#managedIdentityClientId(String)} on the builder or the environment variable AZURE_CLIENT_ID configures the DefaultAzureCredential to authenticate as a user-defined managed identity, while leaving them empty configures it to authenticate as a system-assigned managed identity.</li>
+ *     <li>Setting .tenantId(String) on the builder or the environment variable AZURE_TENANT_ID configures the DefaultAzureCredential to authenticate to a specific tenant for Visual Studio Code, and IntelliJ IDEA.</li>
+ * </ol>
+ *
+ * <p><strong>Sample: Construct DefaultAzureCredential</strong></p>
+ * <!-- src_embed com.azure.identity.credential.defaultazurecredential.construct -->
+ * <pre>
+ * DefaultAzureCredential defaultAzureCredential = new DefaultAzureCredentialBuilder&#40;&#41;
+ *     .build&#40;&#41;;
+ * </pre>
+ * <!-- end com.azure.identity.credential.defaultazurecredential.construct -->
+ *
+ * <p><strong>Sample: Construct DefaultAzureCredential with User Assigned Managed Identity </strong></p>
+ * <!-- src_embed com.azure.identity.credential.defaultazurecredential.constructwithuserassignedmanagedidentity -->
+ * <pre>
+ * DefaultAzureCredential dacWithUserAssignedManagedIdentity = new DefaultAzureCredentialBuilder&#40;&#41;
+ *     .managedIdentityClientId&#40;&quot;&lt;Managed-Identity-Client-Id&quot;&#41;
+ *     .build&#40;&#41;;
+ * </pre>
+ * <!-- end com.azure.identity.credential.defaultazurecredential.constructwithuserassignedmanagedidentity -->
+ *
+ *
+ * @see com.azure.identity.ManagedIdentityCredential
```

Comment:
Include at the top an `{@see com.azure.identity}` to link to the package level javadoc.